src/Icicle/Source/ToCore/ToCore.hs

```diff
@@ -646,16 +580,28 @@ convertReduce xx
 
         scrut' <- convertReduce scrut
         pats'  <- convertCaseFreshenPats pats
-        alts'  <- mapM convertExp alts
 
-        let bs' = fst scrut'
+        -- Because the alternatives can contain more folds and stuff, we need to
+        -- use convertReduce on them.
+        -- However, because the pattern variables must be Aggregates, we know that
+        -- any pattern variables will only be mentioned in the postcomputations.
+        -- Therefore we need to pull out the postcomputations into a let,
+        -- and stick them inside the new case alternative.
+        --
+        -- All the foldy bits of each alternative must be computed, because
+        -- we won't know which ones will be needed until after they are run.
```

Comment:
I agreed with that the postcomputations should be squashed into the one `let`, but I'm not sure I understand what sticking them inside a new alternative means?

Also, after çase desugar, there should be no more pattern variables. They should all become let's, e.g. `case x | y -> ...` should become `let y = x in ...`